tuning/scripts/master_tuning_script.sh

```diff
@@ -1,414 +1,330 @@
+		printf "hgemm detected\n"
+		DATA_TYPE=hgemm
+		DVAL=4
+		if [ "$(grep -c 'compute_type f16_r' "${LOG_PATH}")" -gt 0 ] || \
+        [ "$(grep -c 'compute_type: "f16_r"' "${LOG_PATH}")" -gt 0 ]; then
+			printf "compute_type of f16 detected, disabling HighPrecisionAccumulate\n"
+		fi
+	else
+		printf "Could not detect data type in log file, assuming sgemm\n"
+		DATA_TYPE=sgemm
+		DVAL=2
+	fi
 fi
 
-if [ -z ${SCLK+foo} ]; then
-    rocm-smi 2>&1 | tee rocmsmi.txt
-    SCLK=$(cat rocmsmi.txt | awk 'FNR == 6 {print $4}' | cut -d M -f 1)
-    printf "SCLK: ${SCLK}\n"
-    rm -rf rocmsmi.txt
+# try to determine GPU
+if [ -z ${LIBRARY+x} ]; then
+	rocm_agent_enumerator 2>&1 | tee rae.txt
+	rocminfo 2>&1 | tee rocminfo.txt
+	if [[ "$(grep -c 'gfx900' rae.txt)" -gt 0 ]]; then
+		LIBRARY=vega10
+		if [[ "$(grep -c 'Compute Unit:            56' rocminfo.txt)" -gt 0 ]]; then
+			printf "v340 GPU detected\n"
+		else
+			printf "mi25 GPU detected\n"
+		fi
+	elif [[ "$(grep -c 'gfx906' rae.txt)" -gt 0 ]]; then
+		LIBRARY=vega20
+		if [[ "$(grep -c 'Compute Unit:            60' rocminfo.txt)" -gt 0 ]]; then
+			printf "mi50 GPU detected\n"
+		else
+			printf "mi60 GPU detected\n"
+		fi
+	elif [[ "$(grep -c 'gfx908' rae.txt)" -gt 0 ]]; then
+		printf "arcturus GPU detected\n"
+		LIBRARY=arcturus
+
+		# currently, only sgemm mfma kernels are supported in automation
+		if [[ "${DATA_TYPE}" == sgemm ]]; then
+			MFMA=true
+		fi
+	else
+		printf "Could not detect GPU, assuming vega20\n"
+		LIBRARY=vega20
+	fi
+	rm -rf rae.txt rocminfo.txt
 fi
 
-if [[ "${TENSILE_CLIENT}" != both && "${TENSILE_CLIENT}" != old ]]; then
-    printf "Setting Tensile Client to new\n"
-    TENSILE_CLIENT=new
+# try to determine clock rate
+if [ -z ${SCLK+x} ]; then
+	rocm-smi 2>&1 | tee rocmsmi.txt
+	SCLK=$(<rocmsmi.txt awk 'FNR == 6 {print $4}' | cut -d M -f 1)
+	echo "SCLK: ${SCLK}"
+	rm -rf rocmsmi.txt
 fi
 
-collect_uniques () {
-    strided=${LOGNAME}-strided.sh
-    regular=${LOGNAME}.sh
-    strided2=${LOGNAME}2-strided.sh
-    regular2=${LOGNAME}2.sh
-
-    pushd scripts
-    bash ../../tuning/scripts/unique-rocblas-logs.sh ${strided}
-    bash ../../tuning/scripts/unique-rocblas-logs.sh ${regular}
-    mv unique-${strided} ${strided}
-    mv unique-${regular} ${regular}
-    popd
-
-    pushd scripts2
-    bash ../../tuning/scripts/unique-rocblas-logs.sh ${strided2}
-    bash ../../tuning/scripts/unique-rocblas-logs.sh ${regular2}
-    mv unique-${strided2} ${strided2}
-    mv unique-${regular2} ${regular2}
-    popd
-}
-
-make_packages()
-{
-    if [[ $(ls -A *.deb | wc -c) -eq 0 ]]; then
-        make package
-        cp *.deb ../../../packages/library
-    fi
-    if [[ $(ls -A clients/*.deb | wc -c) -eq 0 ]]; then
-        make package_clients
-        cp clients/*.deb ../../../packages/client
-    fi
-}
-
-
+# run the four individual tuning scripts (and Tensile) in order
+SCRIPT_PATH=$(realpath "${0}" | xargs dirname)
```

Comment:
I don't think realpath is compatible on OSX or CentOS. (unsure about Windows).
If there are real use cases where we need to support OSX, then we might want to think about alternatives for this.

Reply:
After a bit of searching I didn't find a good cross platform way of doing this succinctly. I've left it as is for now, as if we move these scripts into Python (which I want to do eventually), it will not be an issue.